### DISCUSSION TITLE
AWS Lambda Extending State to All Functions

### DISCUSSION THREAD

**Discussion by diegopulpo** [serverless/serverless#10102] [Q&A]
I got this email from Amazon, does it affects the lambdas deployed with serverless?

Hello,

You are receiving this message because we identified that your account created or updated or invoked Lambda functions on or after July 1, 2021.

AWS Lambda is extending the capability to track the current state of a function through its lifecycle to all functions [1]. With this change, you may need to update your CLI and/or SDK-based automation workflows around creating and updating functions by adding a check that the function became active before performing additional actions that operate on the function.

Previously, states have been used in two primary use-cases. Firstly, to move the blocking setup of VPC resources out of the path of function invocation. Secondly, to allow the Lambda service to optimize new or updated container images for container-image based functions, also before invocation. By moving this additional work out of the path of the invocation, customers see lower latency and better consistency in their function performance.
Infrastructure as code and deployment automation tools such as AWS CloudFormation, AWS Serverless Application Model (SAM), AWS Cloud Development Kit (CDK), Serverless Framework,Hashicorp Terraform, AWS Chalice and Cloud Custodian already support states.

If you are using these tools and are already on the latest Software Development Kit (SDK), you do not need to take any action, and can stop reading now. If you are using an earlier SDK version, please update to the latest one.

We are rolling out this change over multi-phase periods to allow you to update existing tooling for deploying and managing Lambda functions. You can delay this change for your functions until December 5, 2021. Starting December 6, 2021, the delay mechanism expires and all customers see the Lambda states lifecycle applied during function create or update. Read this blog post [2] to learn about this change, timelines for different phases and a reference example on how to check your function state.

If you have any questions, please contact AWS Support [3].

[1] https://docs.aws.amazon.com/lambda/latest/dg/functions-states.html
[2] https://aws.amazon.com/blogs/compute/coming-soon-expansion-of-aws-lambda-states-to-all-functions
[3] https://aws.amazon.com/support

Sincerely,
Amazon Web Services

**Comment by mnapoli** [comment]
> Infrastructure as code and deployment automation tools such as AWS CloudFormation, AWS Serverless Application Model (SAM), AWS Cloud Development Kit (CDK), **Serverless Framework**,Hashicorp Terraform, AWS Chalice and Cloud Custodian **already support states.**

😉  but I admit I also struggled to read that part too!

**Reply by kerr-bighealth** [reply]
Does the Serverless Framework version in use have any bearing on the "earlier SDK version" caveat?

> If you are using these tools and are already on the latest Software Development Kit (SDK), you do not need to take any action, and can stop reading now. If you are using an earlier SDK version, please update to the latest one.
